OpenRA.Mods.Common/Traits/Health.cs

```diff
@@ -17,14 +18,27 @@ public class HealthInfo : ITraitInfo, UsesInit<HealthInit>
 	{
 		[Desc("HitPoints")]
 		public readonly int HP = 0;
+		[Desc("Trigger interfaces such as AnnounceOnKill?")]
+		public readonly bool NotifyAppliedDamage = true;
 
-		[Desc("Physical size of the unit used for damage calculations. Impacts within this radius apply full damage.")]
+		[Desc("Shape of the unit used to calculate warhead damage")]
+		public readonly HitShape Shape = HitShape.Circle;
 		public readonly WDist Radius = new WDist(426);
+		public readonly int2[] Vertices = null;
 
-		[Desc("Trigger interfaces such as AnnounceOnKill?")]
-		public readonly bool NotifyAppliedDamage = true;
+		public IHitShape ShapeInstance { get; private set; }
 
-		public virtual object Create(ActorInitializer init) { return new Health(init, this); }
+		public virtual object Create(ActorInitializer init)
+		{
+			switch (Shape)
```

Comment:
Actually, I would have annotate `ShapeInstance` with a custom `[LoadUsing]` and then make the vertices / other info a sub-tree of the field.  You can then drop the Vertices definition here, where it doesn't really belong.